File: coyote/apis/auth_token.py
# This file is @generated

from ..internal.api_common import ApiBase
from ..models import (
    AuthTokenCreateIn,
    AuthTokenCreateOut,
    AuthTokenDeleteIn,
    AuthTokenDeleteOut,
    AuthTokenExpireIn,
    AuthTokenExpireOut,
    AuthTokenListIn,
    AuthTokenRotateIn,
    AuthTokenRotateOut,
    AuthTokenUpdateIn,
    AuthTokenUpdateOut,
    AuthTokenVerifyIn,
    AuthTokenVerifyOut,
    ListResponseAuthTokenOut,
)
from .auth_token_namespace import (
    AuthTokenNamespace,
    AuthTokenNamespaceAsync,
)


class AuthTokenAsync(ApiBase):
    @property
    def namespace(self) -> AuthTokenNamespaceAsync:
        return AuthTokenNamespaceAsync(self._client)

    async def create(
        self,
        auth_token_create_in: AuthTokenCreateIn,
    ) -> AuthTokenCreateOut:
        """Create Auth Token"""
        body = auth_token_create_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.auth-token.create",
            body=body,
            response_type=AuthTokenCreateOut,
        )

    async def expire(
        self,
        auth_token_expire_in: AuthTokenExpireIn,
    ) -> AuthTokenExpireOut:
        """Expire Auth Token"""
        body = auth_token_expire_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.auth-token.expire",
            body=body,
            response_type=AuthTokenExpireOut,
        )

    async def delete(
        self,
        auth_token_delete_in: AuthTokenDeleteIn,
    ) -> AuthTokenDeleteOut:
        """Delete Auth Token"""
        body = auth_token_delete_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.auth-token.delete",
            body=body,
            response_type=AuthTokenDeleteOut,
        )

    async def verify(
        self,
        auth_token_verify_in: AuthTokenVerifyIn,
    ) -> AuthTokenVerifyOut:
        """Verify Auth Token"""
        body = auth_token_verify_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.auth-token.verify",
            body=body,
            response_type=AuthTokenVerifyOut,
        )

    async def list(
        self,
        auth_token_list_in: AuthTokenListIn,
    ) -> ListResponseAuthTokenOut:
        """List Auth Tokens"""
        body = auth_token_list_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.auth-token.list",
            body=body,
            response_type=ListResponseAuthTokenOut,
        )

    async def update(
        self,
        auth_token_update_in: AuthTokenUpdateIn,
    ) -> AuthTokenUpdateOut:
        """Update Auth Token"""
        body = auth_token_update_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.auth-token.update",
            body=body,
            response_type=AuthTokenUpdateOut,
        )

    async def rotate(
        self,
        auth_token_rotate_in: AuthTokenRotateIn,
    ) -> AuthTokenRotateOut:
        """Rotate Auth Token"""
        body = auth_token_rotate_in.model_dump(exclude_none=True)

        return await self._request_asyncio(
            method="post",
            path="/api/v1.auth-token.rotate",
            body=body,
            response_type=AuthTokenRotateOut,
        )


class AuthToken(ApiBase):
    @property
    def namespace(self) -> AuthTokenNamespace:
        return AuthTokenNamespace(self._client)

    def create(
        self,
        auth_token_create_in: AuthTokenCreateIn,
    ) -> AuthTokenCreateOut:
        """Create Auth Token"""
        body = auth_token_create_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.auth-token.create",
            body=body,
            response_type=AuthTokenCreateOut,
        )

    def expire(
        self,
        auth_token_expire_in: AuthTokenExpireIn,
    ) -> AuthTokenExpireOut:
        """Expire Auth Token"""
        body = auth_token_expire_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.auth-token.expire",
            body=body,
            response_type=AuthTokenExpireOut,
        )

    def delete(
        self,
        auth_token_delete_in: AuthTokenDeleteIn,
    ) -> AuthTokenDeleteOut:
        """Delete Auth Token"""
        body = auth_token_delete_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.auth-token.delete",
            body=body,
            response_type=AuthTokenDeleteOut,
        )

    def verify(
        self,
        auth_token_verify_in: AuthTokenVerifyIn,
    ) -> AuthTokenVerifyOut:
        """Verify Auth Token"""
        body = auth_token_verify_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.auth-token.verify",
            body=body,
            response_type=AuthTokenVerifyOut,
        )

    def list(
        self,
        auth_token_list_in: AuthTokenListIn,
    ) -> ListResponseAuthTokenOut:
        """List Auth Tokens"""
        body = auth_token_list_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.auth-token.list",
            body=body,
            response_type=ListResponseAuthTokenOut,
        )

    def update(
        self,
        auth_token_update_in: AuthTokenUpdateIn,
    ) -> AuthTokenUpdateOut:
        """Update Auth Token"""
        body = auth_token_update_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.auth-token.update",
            body=body,
            response_type=AuthTokenUpdateOut,
        )

    def rotate(
        self,
        auth_token_rotate_in: AuthTokenRotateIn,
    ) -> AuthTokenRotateOut:
        """Rotate Auth Token"""
        body = auth_token_rotate_in.model_dump(exclude_none=True)

        return self._request_sync(
            method="post",
            path="/api/v1.auth-token.rotate",
            body=body,
            response_type=AuthTokenRotateOut,
        )

File: src/api/auth_token.rs
// this file is @generated
use super::AuthTokenNamespace;
use crate::{Configuration, error::Result, models::*};

pub struct AuthToken<'a> {
    cfg: &'a Configuration,
}

impl<'a> AuthToken<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    pub fn namespace(&self) -> AuthTokenNamespace<'a> {
        AuthTokenNamespace::new(self.cfg)
    }

    /// Create Auth Token
    pub async fn create(
        &self,
        auth_token_create_in: AuthTokenCreateIn,
    ) -> Result<AuthTokenCreateOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.auth-token.create")
            .with_body(auth_token_create_in)
            .execute(self.cfg)
            .await
    }

    /// Expire Auth Token
    pub async fn expire(
        &self,
        auth_token_expire_in: AuthTokenExpireIn,
    ) -> Result<AuthTokenExpireOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.auth-token.expire")
            .with_body(auth_token_expire_in)
            .execute(self.cfg)
            .await
    }

    /// Delete Auth Token
    pub async fn delete(
        &self,
        auth_token_delete_in: AuthTokenDeleteIn,
    ) -> Result<AuthTokenDeleteOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.auth-token.delete")
            .with_body(auth_token_delete_in)
            .execute(self.cfg)
            .await
    }

    /// Verify Auth Token
    pub async fn verify(
        &self,
        auth_token_verify_in: AuthTokenVerifyIn,
    ) -> Result<AuthTokenVerifyOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.auth-token.verify")
            .with_body(auth_token_verify_in)
            .execute(self.cfg)
            .await
    }

    /// List Auth Tokens
    pub async fn list(
        &self,
        auth_token_list_in: AuthTokenListIn,
    ) -> Result<ListResponseAuthTokenOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.auth-token.list")
            .with_body(auth_token_list_in)
            .execute(self.cfg)
            .await
    }

    /// Update Auth Token
    pub async fn update(
        &self,
        auth_token_update_in: AuthTokenUpdateIn,
    ) -> Result<AuthTokenUpdateOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.auth-token.update")
            .with_body(auth_token_update_in)
            .execute(self.cfg)
            .await
    }

    /// Rotate Auth Token
    pub async fn rotate(
        &self,
        auth_token_rotate_in: AuthTokenRotateIn,
    ) -> Result<AuthTokenRotateOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1.auth-token.rotate")
            .with_body(auth_token_rotate_in)
            .execute(self.cfg)
            .await
    }
}
